pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go

```diff
@@ -422,24 +434,38 @@ func (m *rangefeedMuxer) receiveEventsFromNode(
 		// get stuck in the first place.
 		if timeutil.Now().Before(nextStuckCheck) {
```

Comment:
Unrelatedly to this PR, which is this not `Now().After(nextStuckCheck)` or `!Now().Before(nextStuckCheck`?

It looks like we're checking the stuck-ness all the time before the `nextStuckCheck` timestamp, and don't check after it. If there is a stuck stream and we miss the `nextStuckCheck` timestamp, we won't restart it. Seems like it should be the other way around?

Reply:
Also, since `nextStuckCheck` is updated below, we're always checking this. Except when we accidentally miss it - then we won't ever check it again.

Could this cause the deadlocks we're seeing?

Reply:
Very much related to this PR.... WOW.
I don't know if this is what's causing stuckness... I'm going to fix it now...

Reply:
I don't think we would be restarting them all the time (see `purgeStuckStreams` which seems to be checking the threshold, so most of the time it would return an empty slice), but we would be checking whether the streams are stuck on **each** iteration of this loop. This is wasteful, but it doesn't seem to be incorrect.

However, if we are slow in one of the iterations, and loop around with `Now()` past the `nextStuckCheck`, we would permanently stop checking this, and rangefeeds would never restart. **This** would be problematic.

Reply:
Ack.